model-optimizer/extensions/ops/one_hot.py

```diff
@@ -51,14 +53,15 @@ def infer(node: Node):
         indices_shape = node.in_port(0).data.get_shape()
         assert indices_shape is not None
         dim = indices_shape.size
+        node_name = node.soft_get('name', node.id)
 
         if node.in_port(1).disconnected():  # IR v7 version
-            assert node.has_valid('depth'), 'The node "{}" must have attribute "depth"'.format(node.name)
+            assert node.has_valid('depth'), 'The node "{}" must have attribute "depth"'.format(node_name)
             depth = node.depth
         else:
-            assert_msg = "OneHot `{0}` ({1} input port value) should be scalar: node: `{2}`, {0} value: `{3}`"
+            assert_msg = "OneHot `{0}` (1 input port value) should be scalar: node: `{1}`, {0} value: `{2}`"
```

Comment:
Message states that input should be scalar. Spec states the same. Why do we allow 1D tesnor with one element here?

Reply:
ONNX spec states the same: depth is 0D tensor, but bertsquad-10.onnx is model from official onnx model zoo, and it has OneHot with depth = [2]. We can resolve this case here, and it makes sense to resolve it.

Reply:
The message is not incorrect, depth should be 0D, but we can resolve 1D depth in case it has 1 element.

Reply:
We are not resolving it. We are hiding the issue for now.

Reply:
I reverted changes in this file and made the transformation for reshaping const to scalar.